crates/api_crud/src/user/create.rs

```diff
@@ -219,38 +218,41 @@ impl PerformCrud for Register {
           followers_url: Some(generate_followers_url(&actor_id)?),
           inbox_url: Some(generate_inbox_url(&actor_id)?),
           shared_inbox_url: Some(Some(generate_shared_inbox_url(&actor_id)?)),
+          is_default_community: true,
           ..CommunityForm::default()
         };
-        blocking(context.pool(), move |conn| {
+        let community = blocking(context.pool(), move |conn| {
           Community::create(conn, &community_form)
         })
-        .await??
+        .await??;
+        vec![community]
       }
     };
-
-    // Sign them up for main community no matter what
-    let community_follower_form = CommunityFollowerForm {
-      community_id: main_community.id,
-      person_id: inserted_person.id,
-      pending: false,
-    };
-
-    let follow = move |conn: &'_ _| CommunityFollower::follow(conn, &community_follower_form);
-    blocking(context.pool(), follow)
-      .await?
-      .map_err(|e| LemmyError::from_error_message(e, "community_follower_already_exists"))?;
-
-    // If its an admin, add them as a mod and follower to main
-    if no_admins {
-      let community_moderator_form = CommunityModeratorForm {
-        community_id: main_community.id,
+    for community in default_communities.iter() {
```

Comment:
This loop here is unnecessary, because the code is only executed when the first user signs up. So at this point, only the default community exists and needs to be followed, but there are no other communities yet.

Reply:
From my understanding, this loop will execute for all user registrations, not just the first, as it is where the user is assigned to default communities. Maybe line 245 might be something that only executes on first user, but since default_communities is a vector, we may as well loop through it in that case too, just to handle edge case where site has lost its admins or something.

Reply:
From my understanding, this loop will execute for all user registrations, not just the first, as it is where the user is assigned to default communities. Maybe line 245 might be something that only executes on first user, but since default_communities is a vector, we may as well loop through it in that case too, just to handle edge case where site has lost its admins or something.